### PR TITLE
Speeds up /datum/light_source/proc/update_corners() by 16% or so

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
## About The Pull Request
Just moving `/turf/proc/get_corners()` into its only caller, slightly shuffled. 

+ Also #defining `/proc/GetRedPart()` etc, which should flatten down to a copytext+text2num operation with #51005. Also defining `parse_light_color()` since it has two callers and does something trivial (splits #ff0000 into three luminance vars). 

## Why It's Good For The Game

profile from first commit:
image

## Changelog
:cl: Naksu
code: Lighting corner updates are ever so slightly faster.
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
Holy fucking hell

**Comment by vuonojenmustaturska** [comment]
Latest commit has an extra 6% + unmeasured speedups to /datum/light_source/New(). I forgot to grab a screenshot of the profile though

File: code/modules/lighting/lighting_source.dm [hunk 1]
```diff
@@ -43,12 +43,10 @@
 	light_range = source_atom.light_range
 	light_color = source_atom.light_color
 
-	parse_light_color()
+	PARSE_LIGHT_COLOR(src)
 
 	update()
 
-	return ..()
```
Comment (vuonojenmustaturska) [inline code/modules/lighting/lighting_source.dm:50]: Literally not a thing

**Comment by spessman-007** [comment]
How did you do the profiling?

**Comment by vuonojenmustaturska** [comment]
there's an in-game profiler, you can do simple A:B profiling by doing something like:

```
/proc/thing_to_profile()
  if (prob(50))
    thing_to_profile_old()
  thing_to_profile_new()

/proc/thing_to_profile_old()
// unmodified old proc
/proc/thing_to_profile_new()
// proc with your changes
```

You might need to switch off a var or something for more complex changes

**Comment by spessman-007** [comment]
Didn't know there was a profile, will look into it - thanks!

**Review by AnturK — APPROVED** [review]

**Pull request merged**